Mathlib/NumberTheory/FLT/Three.lean

```diff
@@ -342,6 +364,292 @@ lemma exists_Solution_of_Solution' : ∃ (S₁ : Solution hζ), S₁.multiplicit
     H := H
     hab := hab }, rfl⟩
 
+namespace Solution
+
+lemma a_add_eta_mul_b : S.a + η * S.b = (S.a + S.b) + λ * S.b := by rw [coe_eta]; ring
+
+/-- Given `(S : Solution)`, we have that `λ ∣ (S.a + η * S.b)`. -/
+lemma lambda_dvd_a_add_eta_mul_b : λ ∣ (S.a + η * S.b) :=
+  a_add_eta_mul_b S ▸ dvd_add (dvd_trans (dvd_pow_self _ (by decide)) S.hab) ⟨S.b, by rw [mul_comm]⟩
+
+/-- Given `(S : Solution)`, we have that `λ ∣ (S.a + η ^ 2 * S.b)`. -/
+lemma lambda_dvd_a_add_eta_sq_mul_b : λ ∣ (S.a + η ^ 2 * S.b) := by
+  rw [show S.a + η ^ 2 * S.b = (S.a + S.b) + λ ^ 2 * S.b + 2 * λ * S.b by rw [coe_eta]; ring]
+  exact dvd_add (dvd_add (dvd_trans (dvd_pow_self _ (by decide)) S.hab) ⟨λ * S.b, by ring⟩)
+    ⟨2 * S.b, by ring⟩
+
+/-- Given `(S : Solution)`, we have that `λ ^ 2` does not divide `S.a + η * S.b`. -/
+lemma lambda_sq_not_a_add_eta_mul_b : ¬ λ ^ 2 ∣ (S.a + η * S.b) := by
+  simp_rw [a_add_eta_mul_b, dvd_add_right S.hab, pow_two, mul_dvd_mul_iff_left
+    hζ.zeta_sub_one_prime'.ne_zero, S.hb, not_false_eq_true]
+
+/-- Given `(S : Solution)`, we have that `λ ^ 2` does not divide `S.a + η ^ 2 * S.b`. -/
+lemma lambda_sq_not_dvd_a_add_eta_sq_mul_b : ¬ λ ^ 2 ∣ (S.a + η ^ 2 * S.b) := by
```

Comment:
Can you use the automorphism `η ↦ η ^ 2` to reduce this to the previous statement?

Reply:
I've golfed the proof a little, but I don't think there is an easy way of using `η ↦ η ^ 2`: `S.a` and `S.b` are in `𝓞 K`, not in `ℤ`.

Reply:
Yes, but the automorphism should send solutions to solutions, so one could use the previous lemma for the image (= preimage) of `S` and then apply the automorphism. But it is maybe not so clear that this will save much overall (since one needs some API for the automorphism). So I'm OK with leaving it as is.

Reply:
Yes, I think the API would be longer, and almost surely useless besides this file.